Example new structure

pritkc opened the pull request:
<!--
     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the MOLE Contributing Guide: https://github.com/csrc-sdsu/mole/blob/master/CONTRIBUTING.md
     - 📖 Read the MOLE Code of Conduct: https://github.com/csrc-sdsu/mole/blob/master/CODE_OF_CONDUCT.md
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.
     - 📝 Add license information in the appropriate place. See MATLAB or C++ code for reference.
      
          SPDX-License-Identifier: GPL-3.0-or-later
          © 2008-2024 San Diego State University Research Foundation (SDSURF).
          See LICENSE file or https://www.gnu.org/licenses/gpl-3.0.html for details. 

     NOTE: Pull Requests from forked repositories need to be reviewed by
     a MOLE Team member before any CI builds will run.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.
-->

## What type of PR is this? (check all applicable)

- [X] Refactor
- [] Feature
- [ ] Bug Fix
- [ ] Optimization
- [X] Example
- [ ] Documentation

## Description

## Related Issues & Documents

<!--
For Pull Requests that relate or close an Issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

## Added/updated tests?
_We encourage you to test all code included with MOLE, including examples.

- [ ] Yes
- [ X ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Read Contributing Guide and Code of Conduct

- [ ] I have read and followed the [Contributing Guide](https://github.com/csrc-sdsu/mole/blob/master/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/csrc-sdsu/mole/blob/master/CODE_OF_CONDUCT.md)

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

<img src="https://miro.medium.com/v2/1*VpQb2kbPdj6vXH2pJqi7Qg.gif" width="100" height="100" />

pritkc commented:
[1] Pending the reviews of the files. 
[2] Upon completion of [1], the existing static paths to the examples (i.e., sphinx direct calls) will be replaced.

valeriabarra commented:
Hi @pritkc , thank you for opening the PR.

I would modify the [`examples/index.md`](https://github.com/csrc-sdsu/mole/blob/master/examples/index.md) file with a bit of description of the directory tree. For instance, that we organize examples by PDE-type and then by dimensionality. 

Also, referring to the [libCEED](https://github.com/CEED/libCEED/blob/main/examples/index.md) model, I would also add to the toc the individual subdirectory `index.md` files for our subdirectories. Thank you

valeriabarra commented:
Also, I think we need to move the content out of [`examples/examples.md`](https://github.com/csrc-sdsu/mole/blob/master/examples/examples.md) to the different relevant subdirectories

pritkc commented:
Hi @valeriabarra, Thank you for reviewing these changes thoroughly. Adding a few follow-up questions on quotes. 

> Also, I think we need to move the content out of [examples/examples.md](https://github.com/csrc-sdsu/mole/blob/master/examples/examples.md) to the different relevant subdirectories

[1] Should I be moving individual Markdown examples to their MATLAB host directories? Considering we won't have any single C++ examples that are not available with its MATLAB counterpart. 

> Also, referring to the [libCEED](https://github.com/CEED/libCEED/blob/main/examples/index.md) model, I would also add to the toc the individual subdirectory index.md files for our subdirectories.

[2] Following my question [1], should I assume we're not making any `index.md` directories for C++ directories?

> I would modify the [examples/index.md](https://github.com/csrc-sdsu/mole/blob/master/examples/index.md) file with a bit of description of the directory tree. For instance, that we organize examples by PDE-type and then by dimensionality.


[3] Would it be better if I added that description under `Readme`? For the Sphinx, we will be adding the `index` below the `README`, so both files will be served on the examples homepage.

valeriabarra commented on examples/examples-doc/Elliptic/1D/Elliptic1D.md:
```diff
@@ -0,0 +1,42 @@
+---
+
+This example is implemented in:
+- [MATLAB](../../../../matlab/Elliptic/1D/elliptic1D.m)
+- [C++](../../../../cpp/Elliptic/1D/elliptic1D.cpp)
+
+Additional MATLAB variants of this example with different boundary conditions:
+- [Homogeneous Dirichlet](../../../../matlab/Elliptic/1D/elliptic1DHomogeneousDirichlet.m)
+- [Non-Homogeneous Dirichlet](../../../../matlab/Elliptic/1D/elliptic1DNonHomogeneousDirichlet.m)
+- [Left Dirichlet, Right Neumann](../../../../matlab/Elliptic/1D/elliptic1DLeftDirichletRightNeumann.m)
+- [Left Dirichlet, Right Robin](../../../../matlab/Elliptic/1D/elliptic1DLeftDirichletRightRobin.m)
+- [Left Neumann, Right Neumann](../../../../matlab/Elliptic/1D/elliptic1DLeftNeumannRightNeumann.m)
+- [Left Neumann, Right Robin](../../../../matlab/Elliptic/1D/elliptic1DLeftNeumannRightRobin.m)
+- [Left Robin, Right Robin](../../../../matlab/Elliptic/1D/elliptic1DLeftRobinRightRobin.m)
+- [Periodic Boundary Conditions](../../../../matlab/Elliptic/1D/elliptic1DPeriodicBC.m)
+- [Non-Periodic Boundary Conditions](../../../../matlab/Elliptic/1D/elliptic1DNonPeriodicBC.m)
```
Hi @pritkc , none of these links work. I tried to build locally. Perhaps we have to fix the relative paths

pritkc replied:
Sorry about that. let me get back to you with that.

pritkc replied:
@valeriabarra I've replaced them with GitHub repo links; they might not preview the results, but you may confirm the path.

valeriabarra replied:
Not quite the correct URL 

![image](https://github.com/user-attachments/assets/5f784bff-05f9-446a-a54a-0cfc3de532ea)

I will have more time to review the new structure early next week. 
Now I see that we have two Examples sections.
![Screenshot from 2025-05-09 16-14-00](https://github.com/user-attachments/assets/8fddbe54-9174-4d0b-ad8f-4552b1c581ef)

pritkc replied:
> @valeriabarra I've replaced them with GitHub repo links; they might not preview the results, but you may confirm the path.

@valeriabarra That looks about right to me. As I mentioned, it links to the GitHub `master` repo, which isn't merged with the latest changes yet. The new folder structure has been introduced with this branch, so it will go into effect all together with these changes. 

I've left the dual examples for you to review; the first one uses our original mechanism of copying the examples into the sphinx/source, while the other is directly from the folder present inside the sphinx/source.

valeriabarra replied:
Hi @pritkc I understand you thought keeping both structures would help the review process, but since this is about file organizations (which render the same in the end in the docs) it would be better to actually only see one file tree in the Files Changed tab. Otherwise it is really hard to review. You can please remove the old structure in an isolated, atomic commit, so that if we decide to ever go back to that, it would be easy to revert. Thank you

pritkc replied:
Thank you, @valeriabarra. I've disabled the old dir with commit [[a7d8df8](https://github.com/csrc-sdsu/mole/pull/121/commits/a7d8df82aaf7373443e200e398af61eefcad9af6)]. You'll find only single (the newer dir) upon a fresh pull.

pritkc commented:
@valeriabarra This branch has been tested and is ready for your review.
Changes from the last post review :
1. Retained the original directory structure of matlab and C++ API examples
2. Moved the examples-doc inside the Sphinx 
3. Added new examples; now each category has at least one example for it. 
4. Repaired the linker references; now they link to their master GH repo
5. Improved the TOC tree structure for better visibility of sections. 

Pending -- 
1. Pre-merger squash, there are no isolated commit requirements for this work. You may squash everything if no further improvements are needed.

pritkc commented:
Will close #135 as well

pritkc commented:
@valeriabarra The above feedback and the duplicate `graphviz` have been taken care of with this last push.
